friends/src/components/AddFriend.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+export default class AddFriend extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: '',
+            age: '',
+            email: ''
+        }
+    }
+
+    handleChange = e => {
+        this.setState({
+            ...this.state,
+            [e.target.name]: e.target.value,
+        });
+    }
+
+    render() {
+        return(
+            <form onSubmit={e => this.props.handleSubmit(e, this.state.name, this.state.age, this.state.email)}>
+                <input 
+                    type="text"
+                    name="name"
+                    placeholder="Enter name here."
+                    value={this.state.name} 
+                    onChange={e => this.handleChange(e)}
```

Comment:
```suggestion
                    onChange={this.handleChange}
```
It will already have access to the event without you passing it